Source/Core/Core/HW/CPU.cpp

```diff
@@ -62,6 +66,32 @@ void CPUManager::ExecutePendingJobs(std::unique_lock<std::mutex>& state_lock)
   }
 }
 
+void CPUManager::RunTimer()
+{
+  std::mutex mtx;
+  std::unique_lock<std::mutex> lck(mtx);
+
+  // Steady clock for greater accuracy of timing
+  std::chrono::steady_clock timer;
+  auto prev_time = timer.now();
+
+  while (timer_finish.wait_for(lck, std::chrono::milliseconds(30000)) == std::cv_status::timeout)
+  {
+    const std::string game_id = SConfig::GetInstance().GetGameID();
```

Comment:
I'm pretty sure this is a race. It's probably safer to get that pushed (from `ConfigManager::SetRunningGameMetadata()`, probably?) than to try to periodically poll it.

Reply:
I'm a bit confused on the suggestions here; for the second one, would you lock the game metadata in the getters in ConfigManager.h?

Reply:
@AdmiralCurtiss Sorry for the mention but 1) I selfishly would really like to see this PR get merged it's the number one thing I have wanted to see in Dolphin for years and 2) I think you might have missed @aminoa's question - I think it's a reasonable question.